### DISCUSSION TITLE
Software updates from packagekit vs apt-get

### DISCUSSION THREAD

**Discussion by RedKage** [cockpit-project/cockpit#17360] [Q&A] [answered]
Hello,

I'm new to cockpit.
In the software update section, it is showing me updates.
But I'm confused. Are the versions number shown in this screen the one already installed or the ones that will be installed?

I'm asking, because I recently had a "software update" on the openssl package.
In cockpit, it states it is version 1.1.1n-0+deb10u1

If I do an `apt show openssl` it prints that I have already version 1.1.1n-0+deb10u1 installed....
If then I proceed with the update from cockpit it shows from the logs that it is installing 1.1.1n-0+deb10u1
If I do after that again an `apt show openssl` it prints that I have version 1.1.1n-0+deb10u1 installed....

What is going on? Is there something I am missing?

So it's like,
- Either the software updates are not update but it's like reinstalling the exact same version
- Or it does install something new? But which version and where since apt-get doesn't seem to see a version change

Also where is packagekit pulling the updates from? Which repo?

Thanks, I'm trying to understand how this works

I'm on a Armbian Buster

**Comment by martinpitt** [comment]
Can you please show the output of `apt policy openssl`, `pkcon get-updates`, and a screenshot? It sounds like it treats one and the same version number as an update?

> Also where is packagekit pulling the updates from? Which repo?

The same that apt itself uses, packagekit uses apt libraries internally. I.e. it's using /etc/apt/sources.list and sources.list.d/. `apt policy` will show some details.

**Comment by RedKage** [comment]
I have just upgraded from Buster to Bullseye though...

`apt policy openssl`
```
openssl:
  Installed: 1.1.1n-0+deb11u2
  Candidate: 1.1.1n-0+deb11u2
  Version table:
 *** 1.1.1n-0+deb11u2 500
        500 https://deb.debian.org/debian-security bullseye-security/main arm64 Packages
        100 /var/lib/dpkg/status
     1.1.1n-0+deb11u1 500
        500 https://deb.debian.org/debian bullseye/main arm64 Packages
```

`pkcon get-updates`
```
Command 'pkcon' not found, but can be installed with:
apt install packagekit-tools
```

`apt policy`
```
Package files:
 100 /var/lib/dpkg/status
     release a=now
 500 http://apt.armbian.com bullseye/bullseye-desktop armhf Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=bullseye-desktop,b=armhf
     origin apt.armbian.com
 500 http://apt.armbian.com bullseye/bullseye-desktop arm64 Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=bullseye-desktop,b=arm64
     origin apt.armbian.com
 500 http://apt.armbian.com bullseye/bullseye-utils armhf Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=bullseye-utils,b=armhf
     origin apt.armbian.com
 500 http://apt.armbian.com bullseye/bullseye-utils arm64 Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=bullseye-utils,b=arm64
     origin apt.armbian.com
 500 http://apt.armbian.com bullseye/main armhf Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=main,b=armhf
     origin apt.armbian.com
 500 http://apt.armbian.com bullseye/main arm64 Packages
     release o=Armbian,a=bullseye,n=bullseye,l=Armbian,c=main,b=arm64
     origin apt.armbian.com
 100 https://deb.debian.org/debian bullseye-backports/non-free armhf Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=non-free,b=armhf
     origin deb.debian.org
 100 https://deb.debian.org/debian bullseye-backports/non-free arm64 Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=non-free,b=arm64
     origin deb.debian.org
 100 https://deb.debian.org/debian bullseye-backports/contrib armhf Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=contrib,b=armhf
     origin deb.debian.org
 100 https://deb.debian.org/debian bullseye-backports/contrib arm64 Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=contrib,b=arm64
     origin deb.debian.org
 100 https://deb.debian.org/debian bullseye-backports/main armhf Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=main,b=armhf
     origin deb.debian.org
 100 https://deb.debian.org/debian bullseye-backports/main arm64 Packages
     release o=Debian Backports,a=bullseye-backports,n=bullseye-backports,l=Debian Backports,c=main,b=arm64
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye-updates/main armhf Packages
     release v=11-updates,o=Debian,a=stable-updates,n=bullseye-updates,l=Debian,c=main,b=armhf
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye-updates/main arm64 Packages
     release v=11-updates,o=Debian,a=stable-updates,n=bullseye-updates,l=Debian,c=main,b=arm64
     origin deb.debian.org
 500 https://deb.debian.org/debian-security bullseye-security/main armhf Packages
     release v=11,o=Debian,a=stable-security,n=bullseye-security,l=Debian-Security,c=main,b=armhf
     origin deb.debian.org
 500 https://deb.debian.org/debian-security bullseye-security/main arm64 Packages
     release v=11,o=Debian,a=stable-security,n=bullseye-security,l=Debian-Security,c=main,b=arm64
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/non-free armhf Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=non-free,b=armhf
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/non-free arm64 Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=non-free,b=arm64
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/contrib armhf Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=contrib,b=armhf
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/contrib arm64 Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=contrib,b=arm64
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/main armhf Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=main,b=armhf
     origin deb.debian.org
 500 https://deb.debian.org/debian bullseye/main arm64 Packages
     release v=11.3,o=Debian,a=stable,n=bullseye,l=Debian,c=main,b=arm64
     origin deb.debian.org
Pinned packages:
```

**Comment by martinpitt** [comment]
> Command 'pkcon' not found, but can be installed with:
> apt install packagekit-tools

Can you please do just that, and try the `pkcon get-updates` again?

It seems apt is happy with the installed version, but packagekit seems to think it's somehow an update. Also, do you have a screenshot, please?

**Reply by RedKage** [reply]
Alright I'm gonna do what you are proposing. Hold on. 
As for the screenshot, you want a screenshot of the software update page with the previous applied updates?

**Reply by RedKage** [reply]
Screenshot:
https://ibb.co/0Dfc47t

`pkcon get-updates`
```
Getting updates               [=========================]
Loading cache                 [=========================]
Finished                      [=========================]
There are no updates available at this time.
```

**Comment by martinpitt** [comment, marked as answer]
Oh! So the page does say "System is up to date". openssl is shown in "Update history", i.e. you installed that package yesterday evening. So it's all good.

**Reply by RedKage** [reply]
But that's the thing... I installed it from the Software Update page. Not from apt.
The Software Update installed openssl version 1.1.1n-0+deb10u1 which was already installed.

Next time I have some updates showing up in the Software Update, I will reply here.

**Comment by martinpitt** [comment]
> But that's the thing... I installed it from the Software Update page. Not from apt.

It's same thing in the end. So you mean that when you first saw the issue, openssl was showing in available updates? Yes, please grab a screenshot and `pkcon get-updates` when you see this again. Thanks!

**Reply by RedKage** [reply]
Correct.
On the system openssl was version 1.1.1n-0+deb10u1. And was reported up to date by apt.
But the Software Update page displayed some updates available, among which was an update to openssl version 1.1.1n-0+deb10u1.
So an update to the same version installed was available.

Will do take screens and stuff when this happens again. Thanks for having taken the time to check this out!
Cheers
